go diffs

tac0turtle opened the pull request:
---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

tac0turtle committed:
add go diffs, again

tac0turtle committed:
remove not needed patches

tac0turtle committed:
config changes

tac0turtle committed:
implement types

tac0turtle committed:
store and some state sync changes

Commit:
state sync libs, and crypto

Commit:
add trace support

Commit:
p2p

Commit:
changes

Commit:
rpc and mempool

Commit:
fix and build

Commit:
light protocol changes

Commit:
remove go.work

Commit:
import fix

Commit:
changes

Commit:
generate metrics

evan-forbes reviewed (commented):
nice, I haven't see anything that shouldn't have been added!

looks like you were in the process of fixing the diff in how we're handling data https://github.com/celestiaorg/celestia-core/blob/d81b8307a23b0c843b8469f4ba1c84dca86a6fa8/types/block.go#L1014-L1028

core doesn't have the DataRootHash field, the name of which isn't important, but ofc just that we ensure the da root from app is being used everywhere

tac0turtle commented:
> nice, I haven't see anything that shouldn't have been added!
> 
> looks like you were in the process of fixing the diff in how we're handling data https://github.com/celestiaorg/celestia-core/blob/d81b8307a23b0c843b8469f4ba1c84dca86a6fa8/types/block.go#L1014-L1028
> 
> core doesn't have the DataRootHash field, the name of which isn't important, but ofc just that we ensure the da root from app is being used everywhere

correct, i saw there was overlap with the existing field in the data of field in comet so i added this to not conflict. ill make sure it works as intended

https://github.com/cometbft/cometbft/blob/main/proto/cometbft/types/v1/types.proto#L79
comet doesnt have the type on the proto, but has it on the struct, you introduced it, so i renamed it to not conflict

Commit:
tests

Commit:
fix commit

tac0turtle committed:
rebase (#7)

* rebase

* fix gzip

---------

Co-authored-by: Marko Baricevic <markobaricevic@Markos-MacBook-Pro.local>

Commit:
Merge branch 'v1.x' into marko/go-diffs

Commit:
fix types tests

Commit:
another test fix

Commit:
fixes

kocubinski committed:
use single arity MetricsProvider

julienrbrt committed:
fix double flag registration

Commit:
set tracer

Commit:
fix config

Commit:
skip wal tests and test data race fix

Commit:
remove prints

Commit:
block sync test vectors

Commit:
remove patches

Commit:
fix rpc tests

Commit:
schema not concurrency safe

tac0turtle committed:
bump ci vulncheck go version

tac0turtle committed:
linting

tac0turtle committed:
linting

tac0turtle committed:
test

tac0turtle committed:
test

tac0turtle committed:
fix

tac0turtle committed:
better testing

tac0turtle committed:
schema comment out
